crates/runtime/src/lib.rs

```diff
@@ -431,44 +486,36 @@ impl Runtime {
             }
         }
 
-        tokio::select! {
-            http_res = http_server_future => {
-                match http_res {
-                    Ok(http_res) => http_res.context(UnableToStartHttpServerSnafu),
-                    Err(source) => {
-                        Err(Error::UnableToJoinTask { source })
-                    }
-                }
-             },
-            flight_res = flight_server_future => {
-                match flight_res {
-                    Ok(flight_res) => flight_res.context(UnableToStartFlightServerSnafu),
-                    Err(source) => {
-                        Err(Error::UnableToJoinTask { source })
-                    }
-                }
-            },
-            open_telemetry_res = open_telemetry_server_future => {
-                match open_telemetry_res {
-                    Ok(open_telemetry_res) => open_telemetry_res.context(UnableToStartOpenTelemetryServerSnafu),
-                    Err(source) => {
-                        Err(Error::UnableToJoinTask { source })
-                    }
-                }
-            },
-            pods_watcher_res = async {
-                if let Some(fut) = pods_watcher_future {
-                    fut.await
-                } else {
-                    futures::future::pending().await
-                }
-            } => {
-                pods_watcher_res.context(UnableToInitializePodsWatcherSnafu)
-            },
-            () = shutdown_signal() => {
-                tracing::info!("Goodbye!");
-                Ok(())
-            },
+        // Start Spicepod watcher
```

Comment:
The Spicepod watcher is not technically a server component. To minimize the scope and complexity of changes, the original behavior is being retained. This can be improved next.